.gitignore

```diff
@@ -1,38 +1,10 @@
-### MacOS specific ###
```

Comment:
@maurolepore should we really remove all these lines from .gitignore?

Reply:
yes, please keep at least `.DS_Store` in `.gitignore` see #125

Reply:
@jacobvjk, this PR reverts commits that I did not author. 

![image](https://user-images.githubusercontent.com/5856545/95330729-98ec9b80-086e-11eb-8514-19c1bb58461a.png)

The next step is to re-add this commits but in much smaller steps so we ensure the tests continue to pass along the way. The best person to re-add those commits are the authors of the reverted commits, because they know the intention behind those changes.

Alternatively, I can try brake down all those changes in little bits that make sense to me and ask questions as they arise.